### PR TITLE
Mailgun integration

### PR DESCRIPTION
**Pull request by sof1105**
i've made a service for the integration of mailgun

### CODE REVIEW

**Comment by jterps08** [comment]
Thanks @sof1105 

@austinpapp FYI, I see issues with the changes to "parameters.default.yml" in d721f58

**Comment by sof1105** [comment]
Now i've added the list create member add and member remove service to powerline API

File: backend/src/Civix/ApiBundle/Controller/GroupController.php [hunk 1]
```diff
@@ -69,9 +70,27 @@ public function createGroupAction(Request $request)
...
 
-        $em->persist($group);
-        $em->flush($group);
+        $slugify = new Slugify();
 
+        $groupName = $slugify->slugify($group->getOfficialName());
+
+        $mailgun = $this->get('civix_core.mailgun')->listcreateAction($groupName,$group->getOfficialDescription());
+
+        if($mailgun['http_response_code'] != 200){
```
Comment (austinpapp) [inline backend/src/Civix/ApiBundle/Controller/GroupController.php]: for now this is fine but i'm not 100% ok with knowing that if mailgun goes down for whatever reason (outside of our influences) we can not create a group.

File: backend/src/Civix/CoreBundle/Controller/MailgunController.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<?php
```
Comment (austinpapp) [inline backend/src/Civix/CoreBundle/Controller/MailgunController.php]: is this controller needed? it has hardcoded values and only 1 function

File: backend/app/config/routing.yml [hunk 3]
```diff
@@ -75,4 +75,9 @@ civix_superuser:
 civix_account:
     resource: "@CivixFrontBundle/Controller/AccountController.php"
     prefix:   /account
-    type: annotation 
\ No newline at end of file
+    type: annotation
+
+mailgun_test:
```
Comment (austinpapp) [inline backend/app/config/routing.yml]: why test?

File: backend/src/Civix/CoreBundle/Service/Mailgun/MailgunApi.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+namespace Civix\CoreBundle\Service\Mailgun;
+
+
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Mailgun\Mailgun;
+
+class MailgunApi {
+
+    static $public_key= "pubkey-13f0754196d3299f46d5bc6337cd1ac5";
```
Comment (austinpapp) [inline backend/src/Civix/CoreBundle/Service/Mailgun/MailgunApi.php]: these keys should be pulled from the parameters.yml and not hardcoded for the world to see in github.

**Comment by austinpapp** [comment]
why was parameters.default.yml deleted?
why has permissions changed for various PNGs? Did you need to touch those file? also, why was backend/web/.htaccess perms changed?

File: backend/src/Civix/ApiBundle/Controller/GroupController.php (hunk 1 above)
Comment (sof1105) [inline backend/src/Civix/ApiBundle/Controller/GroupController.php]: when mailgun has an issue it should return an error to the end user and that's why i'm testing if it's not a 200 OK i'll return an error

File: backend/src/Civix/CoreBundle/Controller/MailgunController.php (hunk 2 above)
Comment (sof1105) [inline backend/src/Civix/CoreBundle/Controller/MailgunController.php]: this is a test function as the API require an authentification and so so i've just created for mailgun api tests

File: backend/app/config/routing.yml (hunk 3 above)
Comment (sof1105) [inline backend/app/config/routing.yml]: that is just for test we gonna use the mailgun service class

File: backend/src/Civix/CoreBundle/Service/Mailgun/MailgunApi.php (hunk 4 above)
Comment (sof1105) [inline backend/src/Civix/CoreBundle/Service/Mailgun/MailgunApi.php]: yes you are right on this 

i'll make the changes

**Comment by sof1105** [comment]
that's my fault i've changeded to parameters.yml whithout making a copy

**Pull request merged**